app/models/pipeline_run_stage.rb

```diff
@@ -0,0 +1,256 @@
+      if job_hash['container'] && job_hash['container']['logStreamName']
+        self.job_log_id = job_hash['container']['logStreamName']
+      end
+    else
+      Airbrake.notify("Error for update job status for pipeline run #{id} with error #{stderr}")
+      self.job_status = STATUS_ERROR
+      self.job_status = STATUS_FAILED if stderr =~ /IndexError/ # job no longer exists
+    end
+    save
+  end
+
+  def file_generated_since_run(s3_path)
+    command = "aws s3 ls #{s3_path}"
+    stdout, _stderr, status = Open3.capture3(command)
+    return false unless status.exitstatus.zero?
+    begin
+      s3_file_time = DateTime.strptime(stdout[0..18], "%Y-%m-%d %H:%M:%S")
+      return (s3_file_time > created_at)
+    rescue
+      return nil
+    end
+  end
+
+  def terminate_job
+    command = "aegea batch terminate #{job_id}"
+    _stdout, _stderr, _status = Open3.capture3(command)
+  end
+
+  def set_pipeline_output
+    return if pipeline_run.pipeline_output
+    pipeline_run.pipeline_output = PipelineOutput.new(pipeline_run: pipeline_run,
+                                                      sample: pipeline_run.sample,
+                                                      total_reads: 0,
+                                                      remaining_reads: 0)
+  end
+
+  def sample_output_s3_path
+    pipeline_run.sample.sample_output_s3_path
+  end
+
+  ########### STAGE SPECIFIC FUNCTIONS BELOW ############
+
+  def host_filtering_command
+    script_name = File.basename(IdSeqPipeline::S3_HOST_FILTER_SCRIPT_LOC)
+    sample = pipeline_run.sample
+    file_type = sample.input_files.first.file_type
+    batch_command_env_variables = "INPUT_BUCKET=#{sample.sample_input_s3_path} OUTPUT_BUCKET=#{sample.sample_output_s3_path} " \
+      "FILE_TYPE=#{file_type} ENVIRONMENT=#{Rails.env} DB_SAMPLE_ID=#{sample.id}"
+    if sample.s3_star_index_path.present?
+      batch_command_env_variables += "STAR_GENOME=#{sample.s3_star_index_path} "
+    end
+    if sample.s3_bowtie2_index_path.present?
+      batch_command_env_variables += "BOWTIE2_GENOME=#{sample.s3_bowtie2_index_path} "
+    end
+    batch_command = "aws s3 cp #{IdSeqPipeline::S3_HOST_FILTER_SCRIPT_LOC} .; chmod 755 #{script_name}; " +
+                    batch_command_env_variables + "./#{script_name}"
+    command = "aegea batch submit --command=\"#{batch_command}\" "
+    memory = sample.sample_memory.present? ? sample.sample_memory : Sample::DEFAULT_MEMORY
+    queue =  sample.job_queue.present? ? sample.job_queue : Sample::DEFAULT_QUEUE
+    command += " --storage /mnt=500 --ecr-image idseq --memory #{memory} --queue #{queue} --vcpus 4"
+    command
+  end
+
+  def alignment_command
+    script_name = File.basename(IdSeqPipeline::S3_ALIGNMENT_SCRIPT_LOC)
+    sample = pipeline_run.sample
+    batch_command_env_variables = "INPUT_BUCKET=#{sample.sample_input_s3_path} OUTPUT_BUCKET=#{sample.sample_output_s3_path} " \
+      "ENVIRONMENT=#{Rails.env} DB_SAMPLE_ID=#{sample.id}"
```

Comment:
FYI looking to the future, the way I have the new pipeline scripts, this command will need FASTQ_BUCKET=#{sample.sample_input_s3_path} and INPUT_BUCKET=#{sample.sample_output_s3_path} (input = output from previous stage). OUTPUT_BUCKET can then either remain #{sample.sample_output_s3_path}, or be changed to a third, separate S3 folder. I think it would be better to change it to a separate folder, because then we can avoid downloading all the big earlier STAR/bowtie outputs when we do lazy-run of the gsnap/rapsearch stage. Would be good to discuss these choices